src/Model/Table/SentencesListsTable.php

```diff
@@ -155,33 +155,44 @@ public function getUserChoices($userId, $sentenceId)
             ->notMatching('SentencesSentencesLists', function ($q) use ($sentenceId) {
                 return $q->where(['SentencesSentencesLists.sentence_id' => $sentenceId]);
             })
-            ->select(['id', 'name', 'user_id'])
-            ->order(['name']);
-
-        $listsOfUser = array();
-        $collaborativeLists = array();
-
-        $currentUserId = CurrentUser::get('id');
-        foreach ($results as $result) {
-            $listId = $result['id'];
-            $listName = $result['name'];
-            $userId = $result['user_id'];
+            ->select([
+                'id',
+                'name',
+                'user_id',
+                'is_mine' => 'IF(SentencesLists.user_id = '.$userId.', TRUE, FALSE)',
+                'is_collaborative' => 'IF(SentencesLists.editable_by = "anyone", TRUE, FALSE)',
```

Comment:
`if(condition, true, false)` can be shortened to `(condition)`.

You can also use the query builder like this, to be sure we are not letting any chance for SQL injections:
```php
$query = $this->find();
$query->select([
                'id',
                'name',
                'user_id',
                'is_mine' => $query->newExpr()->eq('SentencesLists.user_id', $userId),
                'is_collaborative' => $query->newExpr()->eq('SentencesLists.editable_by', 'anyone'),
            ]);
```